src/System.Collections/src/System/Collections/Generic/Dictionary.cs

```diff
@@ -249,7 +249,7 @@ private void CopyTo(KeyValuePair<TKey, TValue>[] array, int index)
 
             if (index < 0 || index > array.Length)
             {
-                throw new ArgumentOutOfRangeException("index", SR.ArgumentOutOfRange_NeedNonNegNum);
+                throw new ArgumentOutOfRangeException("index", index, SR.ArgumentOutOfRange_NeedNonNegNum);
```

Comment:
This choice of message is a bug in the original (also present in full framework) in that `((ICollection)new Dictionary<string, int>()).CopyTo(new KeyValuePair<string, int>[3], 100);` complains "Non-negative number required.". If you're going to change it, may as well fix it.